river/cluster/textclust.py

```diff
@@ -0,0 +1,595 @@
+    In: Asian Conference on Intelligent Information and Database Systems (Accepted)
+    [^2]: Carnein, M., Assenmacher, D., Trautmann, H. (2017). Stream Clustering of Chat Messages with Applications to Twitch Streams.
+          In: de Cesare, S., Frank, U. (eds) Advances in Conceptual Modeling. ER 2017.
+
+    Examples
+    --------
+
+    >>> from river import compose
+    >>> from river import feature_extraction
+    >>> from river import metrics
+    >>> from river import cluster
+
+    >>> corpus = [
+    ...    {"text":'This is the first document.',"idd":1, "cluster": 1, "cluster":1},
+    ...    {"text":'This document is the second document.',"idd":2,"cluster": 1},
+    ...    {"text":'And this is super unrelated.',"idd":3,"cluster": 2},
+    ...    {"text":'Is this the first document?',"idd":4,"cluster": 1},
+    ...    {"text":'This is super unrelated as well',"idd":5,"cluster": 2}
+    ... ]
+
+    >>> stopwords = [ 'stop', 'the', 'to', 'and', 'a', 'in', 'it', 'is', 'I', 'that', 'had', 'on', 'for', 'were', 'was']
+
+    >>> metric = metrics.AdjustedRand()
+
+    >>> model = compose.Pipeline(
+    ...     feature_extraction.BagOfWords(lowercase=True, ngram_range=(1,2), stop_words=stopwords),
+    ...     cluster.textclust(realtimefading=False, _lambda=0.001, tgap=100, auto_r=True, radius=0.9)
+    ... )
+
+    >>> for x in corpus:
+    ...     y_pred = model.predict_one(x["text"])
+    ...     y = x["cluster"]
+    ...     metric = metric.update(y,y_pred)
+    ...     model = model.learn_one(x["text"], id = x["idd"])
+
+    >>> print(metric)
+    AdjustedRand: -0.08695652173913043
+
+    """
+
+    # constructor with default specification
+    def __init__(
+        self,
+        radius: float = 0.3,
+        _lambda: float = 0.0005,
+        tgap: int = 100,
+        termfading: bool = True,
+        realtimefading: bool = True,
+        micro_distance: str = "tfidf_cosine_distance",
+        macro_distance: str = "tfidf_cosine_distance",
+        num_macro: int = 3,
+        min_weight: int = 0,
+        auto_r: bool = False,
+        auto_merge: bool = True,
+        sigma: float = 1.0,
+    ):
+
+        self.radius = radius
+        self._lambda = _lambda
+        self.tgap = tgap
+        self.termfading = termfading
+        self.num_macro = num_macro
+        self.realtimefading = realtimefading
+        self.min_weight = min_weight
+        self.auto_r = auto_r
+        self.auto_merge = auto_merge
+        self.sigma = sigma
+
+        # Initialize important values
+        self.num_merged_obs = 0
+        self.t = None
+        self.lastCleanup = 0
+        self.n = 1
+        self.omega = 2 ** (-1 * self._lambda * self.tgap)
+
+        self.assignment: typing.Dict[int, int] = {}
+        self.microclusters: typing.Dict[int, "textclust.microcluster"] = {}
+        self.clusterId = 0
+        self.microToMacro = None
+        self.upToDate = False
+        self.realtime = None
+        self.dist_mean = 0
```

Comment:
Shouldn't a lot of these attributes be private? I don't think you want to expose to users. If such is the case, I would add a leading `_` to mark them as private.